code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -1,265 +1,262 @@
-
-/obj/item/clothing/mask/facehugger/throw_at(atom/target, range, speed, mob/thrower, spin=1, diagonals_first = 0, datum/callback/callback, quickstart = TRUE)
-	. = ..()
-	if(!.)
+	TryCoupling(AM)
+
+/mob/living/simple_animal/hostile/facehugger/attack_hand(mob/user)
+	if(stat == DEAD || sterile)
+		var/obj/item/clothing/mask/facehugger_item/hugger_item = BecomeItem()
+		user.put_in_hands(hugger_item)
+
+/mob/living/simple_animal/hostile/facehugger/attack_alien(mob/user)
+	var/obj/item/clothing/mask/facehugger_item/hugger_item = BecomeItem()
+	user.put_in_hands(hugger_item)
+
+/**
+ * Attempts to have the facehugger couple with the given target.  Checks all possibilities and plays them out accordingly.
+ *
+ * The proc which controls the facehugger's coupling power.  Checks a number of possibilities involving the target, and plays them out.
+ * Smashes against animals, aliens, and cyborgs not in the facehugger's faction.  If the facehugger isn't sterile, this will deal damage to the target.
+ * If the target has no head, it will also bash against them.
+ * If the target wears a helmet which covers their face, the coupling attempt will fail and no damage is dealt.
+ * If the target is not wearing a helmet but instead a removable mask which isn't a facehugger, the facehugger will rip it off.
+ * If the facehugger wasn't stopped anywhere in the previous conditions, it will become an item and attach itself to the target's face, assuming they can wear masks.
+ * Arguments:
+ * * target - the atom which is being checked for coupling
+ */
+/mob/living/simple_animal/hostile/facehugger/proc/TryCoupling(atom/target)
+	if(!istype(target, /mob/living) || stat == DEAD || world.time < couple_timeout)
+		return FALSE
+	if(isanimal(target) || isalien(target) || iscyborg(target))
+		var/mob/living/living_target = target
+		if(faction_check_mob(living_target))
+			return FALSE
+		visible_message("<span class='danger'>[src] smashes against [target], but can't seem to latch on!</span>", "<span class='userdanger'>[src] smashes against you, but can't seem to latch on!</span>")
+		if(!sterile)
+			living_target.adjustBruteLoss(melee_damage_upper)
+			return TRUE
+	if(!iscarbon(target))
 		return
-	if(stat == CONSCIOUS)
-		icon_state = "[initial(icon_state)]_thrown"
-		addtimer(CALLBACK(src, .proc/clear_throw_icon_state), 15)
-
-/obj/item/clothing/mask/facehugger/proc/clear_throw_icon_state()
-	if(icon_state == "[initial(icon_state)]_thrown")
-		icon_state = "[initial(icon_state)]"
-
-/obj/item/clothing/mask/facehugger/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
-	..()
-	if(stat == CONSCIOUS)
-		icon_state = "[initial(icon_state)]"
-		Leap(hit_atom)
+	var/mob/living/carbon/carbon_target = target
 
-/obj/item/clothing/mask/facehugger/proc/valid_to_attach(mob/living/M)
-	// valid targets: carbons except aliens and devils
-	// facehugger state early exit checks
-	if(stat != CONSCIOUS)
+	//Check for immunity
+	if(HAS_TRAIT(carbon_target, TRAIT_XENO_IMMUNE))
```

Comment:
Check for this earlier? Like on the very first line?
That way the trait can be applied to any living.